rust/hyperlane-core/src/traits/mailbox.rs

```diff
@@ -60,3 +71,23 @@ pub trait Mailbox: HyperlaneContract + Send + Sync + Debug {
     /// against the provided signed checkpoint
     fn process_calldata(&self, message: &HyperlaneMessage, metadata: &[u8]) -> Vec<u8>;
 }
+
+/// The result of processing a batch of messages
+#[derive(new, Debug)]
+pub struct BatchResult {
+    /// The outcome of executing the batch, if one was sent
+    pub outcome: Option<TxOutcome>,
+    /// Indexes of excluded calls from the batch (i.e. that were not executed)
+    pub excluded_call_indexes: Vec<usize>,
```

Comment:
nit: wonder if this should be `failed_indexes` bc the exclusion is based on how BatchResult is used and BatchResult isn't really aware of this